src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,309 @@
+    #  - these two lines will replace the for loop with the try/excepts & following two sets
+    #  - temp: rels_mondo_non_source & rels_mondo_non_synced are temporary while these 2 files out of sync
+    #  - perhaps it will be useful to keep the 'non synced check' above and throw an error? but probably not
+    # rels_mondo_source = [(mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]) for x in rels_mondo_mondo]
+    # rels_mondo_source = [x for x in rels_mondo_source if any([x[0].startswith(y) for y in owned_prefix_map]) and
+    #   any([x[2].startswith(y) for y in owned_prefix_map])]
+    rels_mondo_non_source_temp = set()
+    rels_mondo_non_synced_temp = set()
+    # - rels_mondo_non_source_temp: Parent and/or child not from source
+    # - rels_mondo_non_synced_temp: Parent and/or child Mondo ID exists in mondo.db but not mondo.sssom.tsv
+    for x in rels_mondo_mondo:
+        try:
+            rels_mondo_source.add((mondo_source_map[x[0]], x[1], mondo_source_map[x[2]]))
+        except KeyError:
+            try:
+                rels_mondo_non_source_temp.add((mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]))
+            except KeyError:
+                rels_mondo_non_synced_temp.add(x)
+    t1 = datetime.now()
+    print(f'  Completed in {(t1 - t0).seconds} seconds')
+
+    # Determine hierarchy diferences -----------------------------------------------------------------------------------
+    # - find which edges appear in both Mondo and source, or only in one or the other
+    # TODO: set diffs to produce these categories
+    #    1. SCR direct in source and Mondo
+    #      DONE: 3. SCR is direct in the source, but not at all in Mondo
+    #        10. SCR is excluded in Mondo, but direct in source
+    #      5. SCR is direct in Mondo, and not in the source at all.
+    in_both_set: Set[RELATIONSHIP] = rels_source.intersection(rels_mondo_source)
+    in_source_only_set: Set[RELATIONSHIP] = rels_source.difference(rels_mondo_source)
+    in_mondo_only_set: Set[RELATIONSHIP] = rels_mondo_source.difference(rels_source)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    robot_subheader = [{
+        'subject_mondo_id': 'ID',
+        'subject_mondo_label': 'LABEL',
+        'object_mondo_id': 'SC %',
+        'subject_source_id': '',  # >A oboInOwl:source?
+        'object_source_id': '',  # ?
+        'object_mondo_label': '',  # ?
+    }]
+    # - Add metadata to edges (labels, both source/Mondo IDs)
+    in_both: List[Dict] = _add_edge_metadata(in_both_set, source_data_map)
+    in_mondo_only: List[Dict] = _add_edge_metadata(in_mondo_only_set, source_data_map)
+    in_source_only: List[Dict] = _add_edge_metadata(in_source_only_set, source_data_map)
+
+    # Case 3: SCR is direct in the source, but not at all in Mondo
+    df3 = pd.DataFrame(in_source_only)
+    df3 = df3.sort_values(['object_mondo_id'], ascending=[False])
```

Comment:
## NULL values in `SOURCE.subclass.added.robot.tsv` (@matentzn)
@matentzn Respond with decision and I will handle.

This is for _Case 3: SCR is direct in the source, but not at all in Mondo_.

This is because sometimes we have source IDs that have not yet been migrated to mondo, so the `object_mondo_id` and `object_mondo_label` for those records will be null. See THIS COMMENT for explanation.

Current output (ORDO): [ordo.subclass.added.robot.tsv.zip](https://github.com/monarch-initiative/mondo-ingest/files/13314082/ordo.subclass.added.robot.tsv.zip)

@matentzn Decision needed:
For now, I've done 'a', but perhaps you prefer 'b', 'c', or some other solution. Can you choose?
a. Sort this table so that all rows with these null values are at the bottom or top (currently I put at bottom).
b. Create a separate output for these records with null values.
c. Filter them out; just ignore them. No output for these records.

Reply:
Came to decision that we will filter these out. They are captured by migration pipeline.